share/config.fish

```diff
@@ -18,8 +18,22 @@ function __fish_default_command_not_found_handler
 end
 
 if status --is-interactive
+	# Existance of string is a good pre-2.3.0 check. Could also check $FISH_VERSION in the future.
+
+	if not contains "string" (builtin -n)
+		set_color red;
+		echo You appear to be trying to run an old fish binary
+		echo with newer scripts installed into $__fish_datadir
+		echo This is an unsupported configuration.
+		set_color --bold
+		echo THINGS WILL NOT WORK.
+		set_color normal
+		echo You may need to uninstall and reinstall fish.
+		sleep 10
+	end
+
 	# Enable truecolor/24-bit support for select terminals
-	if not set -q NVIM_LISTEN_ADDRESS # Neovim will swallow the 24bit sequences, rendering text white
+	if not set -q NVIM_LISTEN_ADDRESS # (Neovim will swallow the 24bit sequences, rendering text white)
```

Comment:
It momentarily confused me - I think I was wondering for a moment why we would want to render text white. I found it made more sense to me if it said "Because neovim would ... we disable 24-bit color" or just had that fact in parenthesis. I think it's more likely to be read as the reasoning as to why we do something, versus an explanation of what we are doing, that way.